File: src/main/java/com/teamwizardry/librarianlib/api/gui/components/ComponentList.java
package com.teamwizardry.librarianlib.api.gui.components;

import com.teamwizardry.librarianlib.api.gui.GuiComponent;
import com.teamwizardry.librarianlib.math.BoundingBox2D;
import com.teamwizardry.librarianlib.math.Vec2;

/**
 * Created by TheCodeWarrior on 7/31/16.
 */
public class ComponentList extends GuiComponent<ComponentList> {
	
	public ComponentList(int posX, int posY) {
		super(posX, posY);
	}
	
	@Override
	public void drawComponent(Vec2 mousePos, float partialTicks) {
		int y = 0;
		
		for (GuiComponent<?> component : components) {
			component.setPos(new Vec2(component.getPos().x, y));
			BoundingBox2D bb = component.getLogicalSize();
			y = bb.max.yi;
		}
	}
}

File: src/main/java/com/teamwizardry/librarianlib/client/gui/components/ComponentList.kt
package com.teamwizardry.librarianlib.client.gui.components

import com.teamwizardry.librarianlib.client.gui.GuiComponent
import com.teamwizardry.librarianlib.common.util.math.Vec2d
import com.teamwizardry.librarianlib.common.util.vec

/**
 * Created by TheCodeWarrior on 7/31/16.
 */
class ComponentList(posX: Int, posY: Int) : GuiComponent<ComponentList>(posX, posY) {

    override fun drawComponent(mousePos: Vec2d, partialTicks: Float) {
        var y = 0

        for (component in components) {
            component.pos = vec(component.pos.x, y)
            val bb = component.getLogicalSize()
            if (bb != null) y = bb.max.yi
        }
    }
}
